build-tools/packages/build-cli/src/commands/modify/fluid-imports.ts

```diff
@@ -0,0 +1,377 @@
+				for (const importSpecifier of namedImports) {
+					const name = importSpecifier.getName();
+
+					/**
+					 * fullImportSpecifierText includes surrounding text like "type" and whitespace. The surrounding whitespace is
+					 * trimmed, but leading or trailing text like "type" or "as foo" (an alias) is still included. This is the
+					 * string that will be used in the new imports.
+					 *
+					 * This ensures aliases abd individual type-only imports are maintained when rewritten.
+					 */
+					const fullImportSpecifierText = importSpecifier.getFullText().trim();
+					const expectedLevel = getApiLevelForImportName(
+						name,
+						data,
+						/* default */ publicLevel,
+						onlyInternal,
+					);
+
+					log.logIndent(
+						`Found import named: '${fullImportSpecifierText}' (${expectedLevel})`,
+						4,
+					);
+					const newSpecifier =
+						expectedLevel === publicLevel ? moduleName : `${moduleName}/${expectedLevel}`;
+
+					// Track the type-only and regular imports separately. In the second pass through the imports, we'll
+					// create new type-only imports for the ones that were originally type-only. Separate lists is a little
+					// more verbose but easier to reason about.
+					if (isTypeOnly) {
+						if (!newTypeOnlyImports.has(newSpecifier)) {
+							newTypeOnlyImports.set(newSpecifier, new Set());
+						}
+						newTypeOnlyImports.get(newSpecifier)?.add(fullImportSpecifierText);
+					} else {
+						if (!newRegularImports.has(newSpecifier)) {
+							newRegularImports.set(newSpecifier, new Set());
+						}
+						newRegularImports.get(newSpecifier)?.add(fullImportSpecifierText);
+					}
+				}
+			}
+		} /* FIRST PASS */
+
+		// SECOND PASS: Update existing imports and add any missing ones
+		for (const importDeclaration of imports) {
+			// Skip non-Fluid imports
+			if (!isFluidImport(importDeclaration)) {
+				continue;
+			}
+
+			const [moduleName] = parseImport(importDeclaration);
+			const moduleSpecifier = importDeclaration.getModuleSpecifierValue();
+			const isTypeOnly = importDeclaration.isTypeOnly();
+
+			// Skip Fluid imports that aren't in the data file
+			if (!mappingData.has(moduleName)) {
+				continue;
+			}
+
+			// Check if there are supposed to be any new imports from the module specifier
+			const newImportNames = isTypeOnly
+				? newTypeOnlyImports.get(moduleSpecifier)
+				: newRegularImports.get(moduleSpecifier);
+
+			// Since there are no imports from this module specifier, remove it.
+			if (newImportNames === undefined) {
+				importDeclaration.remove();
+				sourceFileChanged = true;
+				continue;
+			}
+
+			// There are new named imports for this module specifier, so remove all individual named imports and immediately
+			// add the new ones.
+			if (newImportNames.size > 0) {
+				importDeclaration.removeNamedImports();
+				importDeclaration.addNamedImports([...newImportNames]);
+				importDeclaration.setIsTypeOnly(isTypeOnly);
```

Comment:
```suggestion
				// We need to set this because we completely removed the declaration earlier, so this is effectively now a new declaration
				importDeclaration.setIsTypeOnly(isTypeOnly);
```